File: Easy/search_insert_position.go
package main

import "fmt"

func searchInsert(nums []int, target int) int {
	index := 0
	for i, n := range nums {
		if n >= target {
			index = i
			break
		}
	}

	if len(nums) > 0 && index == 0 && nums[len(nums)-1] < target {
		return len(nums)
	}

	return index
}

func run_searchInsert() {
	fmt.Println(searchInsert([]int{1, 3, 5, 6}, 5))
}

File: Easy/search_insert_position.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function (nums, target) {
  let index = 0;
  for (let i = 0; i < nums.length; i++) {
    if(nums[i] >= target) {
      index = i;
      break;
    }
  }

  if (nums.length > 0 && index == 0 && nums[nums.length - 1] < target) {
    return nums.length;
  }

  return index;
};

console.log(searchInsert([1, 3, 5, 6], 5));
